src-backbone/app/js/views/builder/builderHeaderView.js

```diff
@@ -11,16 +11,27 @@ module.exports = Marionette.ItemView.extend({
         titleField: 'input#change-title',
         authorField: 'input#change-author',
         downloadButton: 'a#download-btn',
+        saveButton: 'a#save-btn'
     },
 
     events: {
         'keyup @ui.titleField': '_save',
         'keyup @ui.authorField': '_save',
-        'click @ui.downloadButton': '_download'
+        'click @ui.downloadButton': '_download',
+        'click @ui.saveButton':  '_saveProcedure'
     },
 
     _save: _.debounce(function() { this._saveToServer(); }, Config.INPUT_DELAY_BEFORE_SAVE),
 
+    _saveProcedure: function() {
+        this._saveToServer();
+        this.model.pages.forEach(function(page) {
```

Comment:
shouldn't the server just do cascade save instead of us making all these network requests?